site/src/xServices/users/usersXService.ts

```diff
@@ -70,393 +80,455 @@ export type UsersEvent =
+  createMachine(
+    {
+      tsTypes: {} as import("./usersXService.typegen").Typegen0,
+      schema: {
+        context: {} as UsersContext,
+        events: {} as UsersEvent,
+        services: {} as {
+          getUsers: {
+            data: TypesGen.User[]
+          }
+          createUser: {
+            data: TypesGen.User
+          }
+          suspendUser: {
+            data: TypesGen.User
+          }
+          deleteUser: {
+            data: undefined
+          }
+          activateUser: {
+            data: TypesGen.User
+          }
+          updateUserPassword: {
+            data: undefined
+          }
+          updateUserRoles: {
+            data: TypesGen.User
+          }
         },
-        tags: "loading",
       },
-      idle: {
-        on: {
-          GET_USERS: {
-            actions: "assignFilter",
+      predictableActionArguments: true,
+      id: "usersState",
+      initial: "startingPagination",
+      states: {
+        startingPagination: {
+          entry: "assignPaginationRef",
+          always: {
```

Comment:
start in this state to spawn the pagination machine, then go to gettingUsers. Since we always go there on mount, we don't need the `useEffect` kicking it off. The other thing the `useEffect` did was respond to filter changes, but we send events for that now.